zanata-adapter-glossary/src/main/java/org/zanata/adapter/glossary/GlossaryCSVReader.java

```diff
@@ -57,45 +60,39 @@ public GlossaryCSVReader(LocaleId srcLang, int batchSize) {
     }
 
     public List<List<GlossaryEntry>> extractGlossary(Reader reader)
-            throws IOException {
+        throws IOException {
         int entryCount = 0;
-        // TODO replace opencsv with apache-commons-csv (a replacement of
-        // opencsv and is in fedora maintained by someone)
-        // apache-commons-csv is not yet released in maven but has been in
-        // fedora. see http://commons.apache.org/csv/
-        CSVReader csvReader = new CSVReader(reader);
         try {
-            List<List<GlossaryEntry>> glossaries =
-                    new ArrayList<List<GlossaryEntry>>();
-            List<String[]> entries = csvReader.readAll();
-
-            validateCSVEntries(entries);
+            Iterable<CSVRecord> rawRecords = CSVFormat.RFC4180.parse(reader);
+            List<CSVRecord> records = Lists.newArrayList(rawRecords);
 
-            Map<String, Integer> descriptionMap = setupDescMap(entries);
+            validateCSVEntries(records);
+            Map<String, Integer> descriptionMap = setupDescMap(records);
             Map<Integer, LocaleId> localeColMap =
-                    setupLocalesMap(entries, descriptionMap);
+                setupLocalesMap(records, descriptionMap);
 
             LocaleId srcLocale = localeColMap.get(0);
 
             if (!srcLang.equals(srcLocale)) {
                 throw new RuntimeException("input source language '" + srcLang
-                        + "' does not match source language in file '"
-                        + srcLocale + "'");
+                    + "' does not match source language in file '"
+                    + srcLocale + "'");
```

Comment:
You IDE seem to have some formatting issue. Continuation Indent should be set to 8.